### ISSUE TITLE
possible improvement in queue handling

### ISSUE THREAD

**Issue by gryphius** [gryphius/fuglu#37]
Edgaras Lukoševičius reported:
```
I have started digging why is
fuglu top process (CPU bound) in my system.
And what I found is that fuglu is wasting(?) more than half of its time
on futex system call. And what is worse futex syscall often returns with
EAGAIN res=-11 error code (see attachments). You can also see that
select syscall is also taking half of the fuglu processes cpu time, but
at least it is not producing errors (not that I see).
```

my first guess would be that this is related to fuglu's task queue. A quick google showed that using timeouts in queues are possibly a bad idea:
http://blog.codedstructure.net/2011/02/concurrent-queueget-with-timeouts-eats.html

->
replace the current queue.get timeouts with the "poison pill" method mentioned in the article above and see if this reduces CPU usage

**Comment by gryphius** [comment]
Edgaras:
```
Average system load dropped from ~8, to ~4, cpu usage for fuglu dropped 
from 250%+ to ~10-20% with short peaks of 100%, but that is normal.

I still see some futex'es but a sane amount of them 
```

**Comment by gryphius** [comment]
fixed in 7d2a07ebdf50f3a0d1976fa4ed46436364c6ce87
